bpe.py

```diff
@@ -170,7 +171,38 @@ def __call__(self, line, **args):
         :return:
         """
         return tokenize_text(self.merge_table, line, self.bpe_dropout_rate, self.random_generator, **args)
-    
+
+
+class BpeVariationalTokenizer:
+    def __init__(self, bpe_dropout_rates, merge_table, random_seed=None):
+        """
+        :param bpe_dropout_rates: np.array of floats [0, 0) with shape (len(merge_table)
+        :param merge_table: dict [(token_1, token_2)] -> priority
+        """
+
+        self.random_generator = np.random.RandomState(random_seed)
+        self.bpe_dropout_rates = bpe_dropout_rates
+        self.merge_table = merge_table
+
+    def __call__(self, line, **args):
+        """
+        :param line: str
+        :return: tokenized line
+        """
+        drop_item = sps.bernoulli.rvs(p=self.bpe_dropout_rates)
```

Comment:
It will be better to pass random_state directly to `rvs`, the code will be as follows:
```
drop_item = sps.bernoulli.rvs(p=self.bpe_dropout_rates, random_state=self.random_state)
```

Reply:
Done